src/Filters/atoms/DateTimeBlock.tsx

```diff
@@ -178,6 +192,14 @@ const DateTimeBlock : React.FC<IProps> = ({
     }
   }, []);
 
+  const onKeyDownChange = useCallback((event) => {
+    if(event.key==='.' || event.key==='e' || event.key==='-'){
+      event.preventDefault();
+    } else if (event.key==='Backspace'){
+      event.target.name === 'hours' ? setDisplayHours('') : setDisplayMinutes('');
+    }
```

Comment:
This is too aggressive. For example, if I want to type `12` but I accidently type `11`, it's expected that if I hit `backspace` once it will delete the last digit and then I can simply hit `2`. However, the user will then realise that it says `2` and not `12` and will have to start again, which will cause them furstration that we can more easily avoid here.

Reply:
Absolutely agree your perspective and hence I have modified a code. 
Thanks